### PR TITLE
Durable on ACA quickstart

### PR DESCRIPTION
**Pull request by lilyjma**
Notes: 
- Shows how to run Durable in ACA locally and deployed to Azure, using MSSQL backend 
- Use .NET 7 isolated because that's the image used in the existing quickstart 

Next steps
- Current quickstart uses real databases for local development because I'm unsure if Azurite can be run in a local container (SQL can). Would be good to test this. 
- Test .NET 8 isolated image, if there's one

### CODE REVIEW

File: samples/DurableFunctions/.DS_Store
Comment (davidmrdavid) [inline samples/DurableFunctions/.DS_Store:1]: non-blocking nit: we should probably remoe this `.DS_Store` file, which is Mac specific, and should probably not be a part of the repo.

File: samples/DurableFunctions/LocalFunctionProj/LocalFunctionProj.csproj [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>net7.0</TargetFramework>
```
Comment (davidmrdavid) [inline samples/DurableFunctions/LocalFunctionProj/LocalFunctionProj.csproj:3]: non-blocking: any reason not to choose .net8, especially given the M1/M2 issues we've encountered w/ .NET7?

File: samples/DurableFunctions/README.md [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+ENV SQLDB_Connection="<<connection string>>"
+```
+
+Obtain your Azure SQL database's connection string by navigating to the database's blade in the Azure portal. Then, under **Settings**, select **Connection strings** and obtain the ADO.NET connection string. Make sure to provide your password in the template provided.
+
+Below is an example of the portal view for obtaining the Azure SQL connection string:
+
+![An Azure connection string as found in the portal](./media/sql_connection_string_portal.png)
+
+Get your Azure Storage account's connection string by nativating to the storage account in the Azure portal. Then, under **Security + networking**, select **Acess keys**.
```
Comment (davidmrdavid) [inline samples/DurableFunctions/README.md:167]:
```suggestion
Get your Azure Storage account's connection string by navigating to the storage account in the Azure portal. Then, under **Security + networking**, select **Access keys**. 
```

File: samples/DurableFunctions/README.md [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+https://<< your app name >>.northeurope.azurecontainerapps.io
+```
+
+Add the following to the end of the URL:
+
+```
+/api/{your http trigger name}
+```
+
+In the sample code above, the name is `DurableFunctionsOrchestrationCSharp_HttpStart`.
```
Comment (davidmrdavid) [inline samples/DurableFunctions/README.md:315]: nit: I'd also include the explicit URL generated by this app, so `/api/DurableFunctionsOrchestrationCSharp_HttpStart`. That'll make copy-pasting to the browser easier

File: samples/DurableFunctions/README.md [hunk 4]
```diff
@@ -0,0 +1,326 @@
+# Create your first C# Durable Functions on Azure Container Apps
+
+[Durable Functions](https://learn.microsoft.com/azure/azure-functions/durable/durable-functions-overview) is an offering of Azure Functions that lets you write stateful functions in a serverless environment. The extension manages state, checkpoints, and restarts for you.
+
+This quickstart will show you how to create a Durable Function that runs on Azure Container Apps.
```
Comment (davidmrdavid) [inline samples/DurableFunctions/README.md:5]: nit - now that the project has been pushed, should we update the README to say that these instructions yield the sample in `LocalFunctionProj`

**Review by davidmrdavid — COMMENTED** [review]
some non-blocking feedback

File: samples/DurableFunctions/LocalFunctionProj/LocalFunctionProj.csproj (hunk 1 above)
Comment (lilyjma) [reply samples/DurableFunctions/LocalFunctionProj/LocalFunctionProj.csproj:3]: yeah, I kind of mentioned that in the description of the PR. The reason was the original guidance [here](https://github.com/Azure/azure-functions-on-container-apps/tree/main?tab=readme-ov-file#create-the-local-function-project-c-isolated---http-trigger) uses a .NET 7 isolated image. I wasn't sure if the .NET 8 isolated image is ready. To be on the safe side, I chose what's proven to work already. That's why I put testing .NET 8 as the next step.

Comment (lilyjma) [reply samples/DurableFunctions/LocalFunctionProj/LocalFunctionProj.csproj:3]: (I'll try to do it today)

File: samples/DurableFunctions/LocalFunctionProj/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+# To enable ssh & remote debugging on app service change the base image to the one below
+# FROM mcr.microsoft.com/azure-functions/dotnet-isolated:3.0-dotnet-isolated7.0-appservice
+FROM mcr.microsoft.com/azure-functions/dotnet-isolated:4-dotnet-isolated7.0
+ENV AzureWebJobsScriptRoot=/home/site/wwwroot \
+    AzureFunctionsJobHost__Logging__Console__IsEnabled=true
+
+COPY --from=installer-env ["/home/site/wwwroot", "/home/site/wwwroot"]
+ENV WEBSITE_HOSTNAME="localhost:80"
+ENV AzureWebJobsStorage=""
+ENV SQLDB_Connection=""
```
Comment (davidmrdavid) [inline samples/DurableFunctions/LocalFunctionProj/Dockerfile:16]: Should we add some sort of comment reminding users to populate these two?

Comment (davidmrdavid) [reply samples/DurableFunctions/LocalFunctionProj/Dockerfile:16]: Though perhaps they shouldn't do that in the Dockerfile, as that risks pushing the creds. Not sure what's the recommended approach here.

File: samples/DurableFunctions/LocalFunctionProj/LocalFunctionProj.csproj (hunk 1 above)
Comment (davidmrdavid) [reply samples/DurableFunctions/LocalFunctionProj/LocalFunctionProj.csproj:3]: I wouldn't stress about it, .NET 7 is fine :) . Just wondering. If it's easy, let's do it. If it's not, then it's fine.

File: samples/DurableFunctions/README.md [hunk 6]
```diff
@@ -0,0 +1,326 @@
+# Create your first C# Durable Functions on Azure Container Apps
+
+[Durable Functions](https://learn.microsoft.com/azure/azure-functions/durable/durable-functions-overview) is an offering of Azure Functions that lets you write stateful functions in a serverless environment. The extension manages state, checkpoints, and restarts for you.
+
+This quickstart will show you how to create a Durable Function that runs on Azure Container Apps, yielding the same app as the one in the *LocalFunctionProj* directory.
```
Comment (davidmrdavid) [inline samples/DurableFunctions/README.md:5]: ultra nit: a "Durable Function" is sometimes overloaded to mean "an orchestrator", so I prefer to say "DF app", or "orchestrators" explicitly.

```suggestion
This quickstart will show you how to create a Durable Function app that runs on Azure Container Apps, yielding the same app as the one in the *LocalFunctionProj* directory. 
```

File: samples/DurableFunctions/README.md [hunk 7]
```diff
@@ -0,0 +1,326 @@
+# Create your first C# Durable Functions on Azure Container Apps
+
+[Durable Functions](https://learn.microsoft.com/azure/azure-functions/durable/durable-functions-overview) is an offering of Azure Functions that lets you write stateful functions in a serverless environment. The extension manages state, checkpoints, and restarts for you.
+
+This quickstart will show you how to create a Durable Function that runs on Azure Container Apps, yielding the same app as the one in the *LocalFunctionProj* directory. 
+
+> Note: Only the [MSSQL storage provider](https://learn.microsoft.com/azure/azure-functions/durable/durable-functions-storage-providers#mssql) is currently supported for running Durable Functions in Azure Container Apps.
```
Comment (davidmrdavid) [inline samples/DurableFunctions/README.md:7]: Isn't there another 'gotcha' related to auto-scaling?

**Review by davidmrdavid — COMMENTED** [review]
Left some nits. I'll admit, I don't have much bandwidth to give this a thorough read, but at a glance it seems functionally complete. So I won't block, but if you're looking for a more thorough pass, it might be best to get a review from someone else :)

File: samples/DurableFunctions/LocalFunctionProj/LocalFunctionProj.csproj (hunk 1 above)
Comment (raorugan) [reply samples/DurableFunctions/LocalFunctionProj/LocalFunctionProj.csproj:3]: .Net 8 is supported as well. Readme page is updated

**Pull request merged**